controller/readwrite.go

```diff
@@ -165,18 +243,31 @@ func (rw *ReadWrite) ServeAPI(ctx context.Context) error {
 	return api.NewAPI(rw.store, rw.drivers, config.IntVal(rw.conf.Port)).Serve(ctx)
 }
 
-// Single run, render, apply of a unit (task).
+// checkApply runs a task by attempting to render the template and applying the
+// task as necessary.
+//
+// An event is stored:
+//  1. whenever a task errors while executing
+//  2. when a dynamic task successfully executes
+//  3. when a scheduled task successfully executes (regardless of if there
+//     were dependency changes)
 //
-// Stores event data on error or on successful _full_ execution of task.
-// Note: We do not store event when a template has not completely rendered since
-// driver.RenderTemplate() may be called many times per full task execution
-// since there could be many per full task execution i.e. when driver.RenderTemplate()
-// returns false, no event is stored.
-func (rw *ReadWrite) checkApply(ctx context.Context, d driver.Driver, retry bool) (bool, error) {
+// Note on #2: no event is stored when a dynamic task renders but does not apply.
+// This can occur becauser driver.RenderTemplate() may need to be called multiple
+// times before a template is ready to be applied.
+func (rw *ReadWrite) checkApply(ctx context.Context, d driver.Driver, retry, once bool) (bool, error) {
 	task := d.Task()
 	taskName := task.Name()
 	if !task.IsEnabled() {
-		rw.logger.Trace("skipping disabled task", taskNameLogKey, taskName)
+		if _, ok := task.Condition().(*config.ScheduleConditionConfig); ok {
```

Comment:
Thanks for asking! I was thinking about this as well and thought about waiting to see if we would have other types of tasks (beyond dynamic and scheduled) before deciding on a method.

But, to your point, this check happens a lot 🤔 and is worth implementing now. I'm thinking of maybe adding a super basic `task.IsScheduled()` to replace this check. It would work for now and then we could evolve it if we have more types of tasks down the road. Let me know if you have thoughts!

Reply:
Makes sense to be thinking of future types! That sounds like a good plan too, where we can revisit once a new type comes up.